<h3>DATA ANALYTICS ON AMAZON TOP RATED SMARTPHONE ACCESSORIES IN 2021 </H3>

**<h2>Introduction</h2>**

Amazon is an American multinational Company that focuses on e-commerce, cloud computing, digital streaming , and Artifical intelligence.It is one of the world's largest retailers; an online go to shop you can get anything you can imagine. Amazon was founded by Jeff Bezos from his garage  on July 5, 1994 in Bellevue, Washignthon where it was Initially an online marketplace for books, but has now expanded into a multitude of product categories: a strategy that has earned it the moniker as "The Everything Store".

**Data Analysis Goal**


Our goal is to answer the following questions from our analysis
<ul>
    <li>Find the Average actual price value, offer value and review count of each phone model</li>
    <li> Which Smartphone model/Acessories has the highest ratings?</li>
    <li> Which SmartPhone Model/Accesories has the highest Review count</li>
 </ul>  

**Business Questions this Analysis will help to solve**

<ol>
 <li> How could these Rating influence the sales of each smartphone</li>
   <li>4.How can your insight drive business decisions</li>
  </ol>

**<h2>SOURCE OF DATA:</h2>**

This dataset was scraped from Amazon e-commerce website ,Amazon.in and was made available at kaggle https://www.kaggle.com/ankitkalauni/amazon-top-rated-smartphones-accessories-2021 on October,2021. This dataset is accessible to everyone for freee download.

In [8]:
#First, we shall Import my libraries

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
from sklearn.impute import SimpleImputer





In [6]:
#Get my first five rows
amazon_df = pd.read_csv("Amazon Top Rated Phone Accessories.csv")
amazon_df.head()

Smartphone              Rating  \
0  Redmi 9 (Sky Blue, 4GB RAM, 64GB Storage) | 2....  4.2 out of 5 stars   
1  Redmi 9A (Nature Green, 2GB RAM, 32GB Storage)...  4.2 out of 5 stars   
2  Redmi 9A (Sea Blue 2GB RAM 32GB Storage) | 2GH...  4.2 out of 5 stars   
3  Redmi Note 10 (Aqua Green, 4GB RAM, 64GB Stora...  4.2 out of 5 stars   
4  Redmi 9 (Carbon Black, 4GB RAM, 64GB Storage) ...  4.2 out of 5 stars   

  Reviews count Offer price Actual price             Saving  \
0        87,809      ₹9,499      ₹10,999  Save ₹1,500 (14%)   
1        74,587      ₹6,999       ₹8,499  Save ₹1,500 (18%)   
2        74,587      ₹6,999       ₹8,499  Save ₹1,500 (18%)   
3        95,519     ₹13,999      ₹15,999  Save ₹2,000 (13%)   
4        87,809      ₹9,499      ₹10,999  Save ₹1,500 (14%)   

                  Delivery Status of stock & more offers  
0  FREE Delivery by Amazon                           NaN  
1  FREE Delivery by Amazon                           NaN  
2  FREE Delivery by Amazon                           NaN  
3  FREE Delivery by Amazon                           NaN  
4  FREE Delivery by Amazon                           NaN

In [9]:
#Next, get the last five rows to have an overview of my dataset
amazon_df.tail(54)

Smartphone              Rating  \
1314  Nillkin for Samsung Galaxy S21 Case, CamShield...  4.6 out of 5 stars   
1315  Nillkin Compatible with Case Samsung Galaxy S2...  4.1 out of 5 stars   
1316  Radhy Selling MBO M10 2.4" Display Dual Sim Mo...  5.0 out of 5 stars   
1317  Nillkin Case for Samsung Galaxy M51 M 51 (6.7"...  5.0 out of 5 stars   
1318  nillkin super frosted shield pro back cover ca...  5.0 out of 5 stars   
1319  Nillkin Compatible with Case iPhone 12 Pro Max...  4.7 out of 5 stars   
1320  Nillkin Case for Huawei P40 Pro (6.58" Inch) Q...  5.0 out of 5 stars   
1321  Nillkin [ Medley] Compatible with iPhone 12 Ca...  5.0 out of 5 stars   
1322  Nillkin Samsung Galaxy Note 20 Ultra Case, Cam...  4.0 out of 5 stars   
1323  Nillkin Polycarbonate iPhone 12 Pro max case -...  4.3 out of 5 stars   
1324  Nillkin Case for Huawei Mate 30 Pro/Mate 30 Pr...  5.0 out of 5 stars   
1325  Nillkin Case for Samsung Galaxy A71 A 71 (6.7"...  5.0 out of 5 stars   
1326  SPARKLE® Set of 12Db 2g/3g/4g Log Periodic Dip...  4.0 out of 5 stars   
1327  Nillkin iPhone SE 2020 Camera Protection Case ...  5.0 out of 5 stars   
1328  Nillkin iPhone 11 Pro Max Case Slide Camera Pr...  4.0 out of 5 stars   
1329  Nillkin Plastic Armor iPhone 11 Pro Max Case, ...  4.5 out of 5 stars   
1330  Nillkin Case for Samsung Galaxy A51 A 51 Super...  5.0 out of 5 stars   
1331  Nillkin Case for Apple iPhone 11 Pro (5.8" Inc...  4.5 out of 5 stars   
1332  Nillkin Case for Apple iPhone 11 Pro (5.8" Inc...  4.0 out of 5 stars   
1333  Nillkin Case for Apple iPhone 11 Pro (5.8" Inc...  5.0 out of 5 stars   
1334  Ghostek Covert Designed for Galaxy Note 10 Plu...  4.5 out of 5 stars   
1335  Nillkin Screen Guard for Google Pixel 3A 3 A W...  4.0 out of 5 stars   
1336  Nillkin Amazing CP+PRO Full Cover Tempered Gla...  4.0 out of 5 stars   
1337  MobiSpare® LCD Display with Touch Screen Combo...  4.2 out of 5 stars   
1338  Nillkin India Super Frosted Shield Hard Back C...  4.0 out of 5 stars   
1339  Nillkin Case for Samsung Galaxy S10e S 10e (5....  5.0 out of 5 stars   
1340                      Vivo V11 (Starry Night Black)  5.0 out of 5 stars   
1341  Nillkin Case for Samsung Galaxy J6 Plus/J 6 Pr...  5.0 out of 5 stars   
1342  Nillkin Screen Guard for Google Pixel 3 XL Sup...  4.6 out of 5 stars   
1343  Nillkin Case for Apple iPhone Xs Max (6.5" Inc...  5.0 out of 5 stars   
1344  Nillkin 3D XD CP+ Max 0.1 mm Thin Edge Shatter...  5.0 out of 5 stars   
1345  nillkin tempered glass for samsung galaxy note...  5.0 out of 5 stars   
1346                 Comio S1 Lite (Sunrise Gold, 32GB)  4.7 out of 5 stars   
1347  Riviera Fast Usb Charger Premium Quality (Bis ...  5.0 out of 5 stars   
1348    Nite Ize Wraptor, Rotating Smartphone Bar Mount  4.6 out of 5 stars   
1349                     Ziox Thunder Mega (Black-Grey)  5.0 out of 5 stars   
1350  VRS Design Case for Apple iPhone 8 Plus (5.5" ...  5.0 out of 5 stars   
1351  Ziox Zelfie Dual SIM & Dual Camera With Front ...  4.0 out of 5 stars   
1352            Itel it5320 Phone with 2.8-inch Display  4.0 out of 5 stars   
1353  i-Smart IS-100i-(Grey+Black) Basic Mobile Phon...  4.5 out of 5 stars   
1354  India Gadgets - Armor X5 Pro Rugged Android Mo...  5.0 out of 5 stars   
1355  Nite Ize S-Biner Marine SlideLock #3-316 Stain...  4.7 out of 5 stars   
1356  (Renewed) Oppo Reno 6 Pro 5G (Aurora, 12GB RAM...  5.0 out of 5 stars   
1357  (Renewed) Ziox X73 with Vibration 1.8" Display...  4.0 out of 5 stars   
1358  New Apple iPhone 12 (256GB) - Blue with Apple ...  4.6 out of 5 stars   
1359  New Apple iPhone 12 (64GB) - Black with Apple ...  4.6 out of 5 stars   
1360  New Apple iPhone 12 (64GB) - Blue with Apple C...  4.6 out of 5 stars   
1361  Lava Z4 Flame Red, 4GB RAM, 64GB Storage + Lav...  4.0 out of 5 stars   
1362  Lava Z4 Flame Red, 4GB RAM, 64GB Storage + Lav...  4.0 out of 5 stars   
1363  Nillkin Case for Apple iPhone 12 / iPhone 12 P...  5.0 out of 5 stars   
1364  Nillkin Ca

In [10]:
#viweing the columns available in my dataset
amazon_df.columns

Index(['Smartphone', 'Rating', 'Reviews count', 'Offer price', 'Actual price',
       'Saving', 'Delivery', 'Status of stock & more offers'],
      dtype='object')

<h3>Column Names</h3>

 `Smartphone` - model name along with its specification
  
   `Ratings` -  This shows the “average customer review" of an item on the scale of 1-5. with 5 being the most desired Smartphone
   
   `Reviews count` - The  total number of reviews each product has
   
   `Offer price` - Price the product was sold for
   
   `Actual price` -The official price for each product
   
   `Savings` - The difference between the Offer price and the actual price
   
   `Delivery` -  Free Deliveries made by Amazon

   `Status of stock & more offers` - Number of product remaning in the inventory
   

<h1>Data Types in our columns</h1>

In [11]:
amazon_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1368 entries, 0 to 1367
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   Smartphone                     1368 non-null   object
 1   Rating                         1368 non-null   object
 2   Reviews count                  1368 non-null   object
 3   Offer price                    1154 non-null   object
 4   Actual price                   1042 non-null   object
 5   Saving                         1151 non-null   object
 6   Delivery                       685 non-null    object
 7   Status of stock & more offers  498 non-null    object
dtypes: object(8)
memory usage: 85.6+ KB


They are only string columns in our dataset

<H3>Number of rows and columns in our Dataset</H3

In [12]:
amazon_df.shape

(1368, 8)

Our Dataset has 1368 rows and 8 columns

<H1>Checking for missing values </H1>

In [13]:
#checking for missing values
amazon_df.isnull().sum() 

Smartphone                         0
Rating                             0
Reviews count                      0
Offer price                      214
Actual price                     326
Saving                           217
Delivery                         683
Status of stock & more offers    870
dtype: int64

The `Status of stock & more offers` column has the highest number of misisng values. Also, `Offer price`, `Actual price`, `saving` and `Delivery`  columns
has Missing values

<h3>No Of Unique values in each Column </h3>

In [14]:
amazon_df.nunique(dropna=True)

Smartphone                       1308
Rating                             11
Reviews count                     363
Offer price                       511
Actual price                      281
Saving                            612
Delivery                            4
Status of stock & more offers      17
dtype: int64

**<h1>Cleaning Steps:</h1>**
<ul>
    <li> Rename columns</li>
    <li> Fix and format the ratings,savings, reviews count, offer price, actual price columns</li>
     <li>Fill in Missing values</li>
    <li>Convert the strings columns to Numerical columns</li>
 </ul>

**RENAMING COLUMNS**

In [19]:
# Renaming the Columns in Dataset in a suitable Pythonic way
amazon_df= amazon_df.rename(columns = {'Smartphone': 'PhoneModel', 'Reviews count': 'ReviewsCount',
                                       'Offer price': 'OfferPrice', 'Actual price': 'ActualPrice',
                                       'Status of stock & more offers': 'StatusOfStock'},inplace= False)

In [20]:
amazon_df.head(50)

PhoneModel              Rating  \
0   Redmi 9 (Sky Blue, 4GB RAM, 64GB Storage) | 2....  4.2 out of 5 stars   
1   Redmi 9A (Nature Green, 2GB RAM, 32GB Storage)...  4.2 out of 5 stars   
2   Redmi 9A (Sea Blue 2GB RAM 32GB Storage) | 2GH...  4.2 out of 5 stars   
3   Redmi Note 10 (Aqua Green, 4GB RAM, 64GB Stora...  4.2 out of 5 stars   
4   Redmi 9 (Carbon Black, 4GB RAM, 64GB Storage) ...  4.2 out of 5 stars   
5   Redmi 9 Power (Blazing Blue, 4GB RAM, 64GB Sto...  4.2 out of 5 stars   
6   Redmi 9A (Midnight Black, 2GB RAM, 32GB Storag...  4.3 out of 5 stars   
7   OnePlus Nord 2 5G (Blue Haze, 8GB RAM, 128GB S...  4.3 out of 5 stars   
8   OnePlus Nord CE 5G (Charcoal Ink, 8GB RAM, 128...  4.1 out of 5 stars   
9   Redmi 9 Prime (Space Blue, 4GB RAM, 64GB Stora...  4.3 out of 5 stars   
10  Redmi Note 10S (Deep Sea Blue, 6GB RAM, 64GB S...  4.1 out of 5 stars   
11  Samsung Galaxy M31 (Ocean Blue, 8GB RAM, 128GB...  4.3 out of 5 stars   
12  Redmi 9 (Sporty Orange, 4GB RAM, 64GB Storage)...  4.2 out of 5 stars   
13  Samsung Galaxy M21 2021 Edition (Arctic Blue, ...  4.2 out of 5 stars   
14  Redmi Note 10 Pro (Dark Night, 6GB RAM, 128GB ...  4.2 out of 5 stars   
15  OnePlus Nord CE 5G (Blue Void, 8GB RAM, 128GB ...  4.1 out of 5 stars   
16  Redmi 9 Power (Electric Green, 4GB RAM, 64GB S...  4.2 out of 5 stars   
17  Redmi Note 10 Pro (Glacial Blue, 6GB RAM, 128G...  4.2 out of 5 stars   
18  Redmi 9A (Sea Blue 3GB RAM 32GB Storage)| 2GHz...  4.2 out of 5 stars   
19                         NOKIA 105 Dual SIM (Black)  4.0 out of 5 stars   
20   Redmi Note 9 Shadow Black, 4GB RAM, 64GB Storage  4.3 out of 5 stars   
21  Redmi 9 Prime (Mint Green, 4GB RAM, 64GB Stora...  4.3 out of 5 stars   
22  Redmi 9 Prime (Matte Black, 4GB RAM, 64GB Stor...  4.3 out of 5 stars   
23  OnePlus Nord 2 5G (Gray Sierra, 8GB RAM, 128GB...  4.3 out of 5 stars   
24  realme C11 (2021) (Cool Grey, 2GB RAM, 32GB St...  4.1 out of 5 stars   
25  Samsung Galaxy M21 2021 Edition (Charcoal Blac...  4.2 out of 5 stars   
26  Redmi 9 Power (Mighty Black 4GB RAM 64GB Stora...  4.2 out of 5 stars   
27  Redmi 9 Power (Blazing Blue, 6GB RAM, 128GB St...  4.2 out of 5 stars   
28  Oppo A31 (Fantasy White, 6GB RAM, 128GB Storag...  4.2 out of 5 stars   
29  OnePlus Nord CE 5G (Charcoal Ink, 6GB RAM, 128...  4.1 out of 5 stars   
30                       Nokia 105 Single SIM (Black)  4.1 out of 5 stars   
31  Redmi Note 10S (Shadow Black, 6GB RAM, 64GB St...  4.1 out of 5 stars   
32  Redmi Note 10S (Shadow Black, 6GB RAM, 128GB S...  4.1 out of 5 stars   
33  OnePlus Nord 2 5G (Green Woods, 12GB RAM, 256G...  4.3 out of 5 stars   
34  Redmi Note 10 Pro (Vintage Bronze, 6GB RAM, 12...  4.2 out of 5 stars   
35  Tecno Spark 7T(Jewel Blue, 4GB RAM, 64GB Stora...  4.0 out of 5 stars   
36  Redmi Note 10 Pro Max (Dark Night, 6GB RAM, 12...  4.1 out of 5 stars   
37  Samsung Galaxy M12 (Blue,4GB RAM, 64GB Storage...  4.1 out of 5 stars   
38  Redmi Note 10 Pro Max (Glacial Blue, 6GB RAM, ...  4.1 out of 5 stars   
39  Redmi Note 10 Pro Max (Vintage Bronze, 6GB RAM...  4.1 out of 5 stars   
40  Samsung Galaxy M32 (Light Blue, 6GB RAM, 128GB...  4.2 out of 5 stars   
41  Redmi 9 Prime (Sunrise Flare, 4GB RAM, 64GB St...  4.3 out of 5 stars   
42  Oppo A31 (Mystery Black, 6GB RAM, 128GB Storag...  4.2 out of 5 stars   
43  OnePlus Nord 2 5G (Gray Sierra, 12GB RAM, 256G...  4.3 out of 5 stars   
44  OnePlus Nord 2 5G (Blue Haze, 12GB RAM, 256GB ...  4.3 out of 5 stars   
45  Samsung Galaxy M21 2021 Edition (Arctic Blue, ...  4.2 out of 5 stars   
46  Samsung Galaxy M32 (Black, 6GB RAM, 128GB Stor...  4.2 out of 5 stars   
47  OPPO A31 (Fantasy White, 4GB RAM, 64GB Storage...  4.2 out of 5 stars   
48  iQOO Z3 5G (Cyber Blue, 8GB RAM, 128GB Storage...  4.3 out of 5 stars   
49  Redmi Note 10S (Frost White, 6GB RAM, 64GB Sto...  4.1 out of 5 stars   

   ReviewsCount OfferPrice ActualPrice                   Saving  \
0        87,809     ₹9,499     ₹10,999        Sa

<h2>Fix and format the ratings,savings, reviews count, offer price, actual price columns</h2>

In [21]:
#Further Data cleaning. Here we will remove the commas & currency signs in the ReviewsCount, OfferPrice and ActualPrice column.
amazon_df['ReviewsCount'] = amazon_df['ReviewsCount'].str.replace(",","")
amazon_df['OfferPrice'] = amazon_df['OfferPrice'].str.replace(",","")
amazon_df['OfferPrice'] = amazon_df['OfferPrice'].str.replace("₹","")
amazon_df['ActualPrice'] = amazon_df['ActualPrice'].str.replace(",","")
amazon_df['ActualPrice'] = amazon_df['ActualPrice'].str.replace("₹","")
amazon_df.head()

PhoneModel              Rating  \
0  Redmi 9 (Sky Blue, 4GB RAM, 64GB Storage) | 2....  4.2 out of 5 stars   
1  Redmi 9A (Nature Green, 2GB RAM, 32GB Storage)...  4.2 out of 5 stars   
2  Redmi 9A (Sea Blue 2GB RAM 32GB Storage) | 2GH...  4.2 out of 5 stars   
3  Redmi Note 10 (Aqua Green, 4GB RAM, 64GB Stora...  4.2 out of 5 stars   
4  Redmi 9 (Carbon Black, 4GB RAM, 64GB Storage) ...  4.2 out of 5 stars   

  ReviewsCount OfferPrice ActualPrice             Saving  \
0        87809       9499       10999  Save ₹1,500 (14%)   
1        74587       6999        8499  Save ₹1,500 (18%)   
2        74587       6999        8499  Save ₹1,500 (18%)   
3        95519      13999       15999  Save ₹2,000 (13%)   
4        87809       9499       10999  Save ₹1,500 (14%)   

                  Delivery StatusOfStock  
0  FREE Delivery by Amazon           NaN  
1  FREE Delivery by Amazon           NaN  
2  FREE Delivery by Amazon           NaN  
3  FREE Delivery by Amazon           NaN  
4  FREE Delivery by Amazon           NaN

**<h3>Filling the Mising values</h3>**

In [22]:
new_amazondf = amazon_df.copy()
new_amazondf['Delivery']=new_amazondf['Delivery'].fillna(new_amazondf['Delivery'].mode()[0])
new_amazondf['StatusOfStock']=new_amazondf['StatusOfStock'].fillna(new_amazondf['StatusOfStock'].mode()[0])
new_amazondf=new_amazondf.fillna(0)
new_amazondf.tail(12)


PhoneModel              Rating  \
1356  (Renewed) Oppo Reno 6 Pro 5G (Aurora, 12GB RAM...  5.0 out of 5 stars   
1357  (Renewed) Ziox X73 with Vibration 1.8" Display...  4.0 out of 5 stars   
1358  New Apple iPhone 12 (256GB) - Blue with Apple ...  4.6 out of 5 stars   
1359  New Apple iPhone 12 (64GB) - Black with Apple ...  4.6 out of 5 stars   
1360  New Apple iPhone 12 (64GB) - Blue with Apple C...  4.6 out of 5 stars   
1361  Lava Z4 Flame Red, 4GB RAM, 64GB Storage + Lav...  4.0 out of 5 stars   
1362  Lava Z4 Flame Red, 4GB RAM, 64GB Storage + Lav...  4.0 out of 5 stars   
1363  Nillkin Case for Apple iPhone 12 / iPhone 12 P...  5.0 out of 5 stars   
1364  Nillkin Case for Apple iPhone 12 / iPhone 12 P...  5.0 out of 5 stars   
1365  Nillkin Case for Samsung Galaxy F62 F 62 / Sam...  5.0 out of 5 stars   
1366  Nillkin Case for Xiaomi Redmi Note 10 / Note 1...  5.0 out of 5 stars   
1367  Nillkin Case for Samsung Galaxy A72 A 72 (6.7"...  5.0 out of 5 stars   

     ReviewsCount OfferPrice ActualPrice                   Saving  \
1356            1          0           0                        0   
1357            1        710         849          Save ₹139 (16%)   
1358         4260     101600           0  FREE Delivery by Amazon   
1359         4260      79199       84800         Save ₹5,601 (6%)   
1360         4260      79199       84800         Save ₹5,601 (6%)   
1361          353       9648       10798        Save ₹1,150 (10%)   
1362          353      10451       11998        Save ₹1,547 (12%)   
1363            1       4499        5999        Save ₹1,500 (25%)   
1364            1       3999        4999        Save ₹1,000 (20%)   
1365            1       1099        1699          Save ₹600 (35%)   
1366            1       1299        1699          Save ₹400 (24%)   
1367            1       1199        2499        Save ₹1,300 (52%)   

                     Delivery          StatusOfStock  
1356  FREE Delivery by Amazon  Only 1 left in stock.  
1357  FREE Delivery by Amazon  Only 1 left in stock.  
1358  FREE Delivery by Amazon  Only 1 left in stock.  
1359  FREE Delivery by Amazon  Only 1 left in stock.  
1360  FREE Delivery by Amazon  Only 1 left in stock.  
1361  FREE Delivery by Amazon  Only 1 left in stock.  
1362  FREE Delivery by Amazon  Only 1 left in stock.  
1363  FREE Delivery by Amazon  Only 1 left in stock.  
1364  FREE Delivery by Amazon  Only 2 left in stock.  
1365  FREE Delivery by Amazon  Only 1 left in stock.  
1366  FREE Delivery by Amazon  Only 1 left in stock.  
1367  FREE Delivery by Amazon  Only 2 left in stock.

**<h3>Convert the strings columns to Numerical columns</h3>**

Convert`ReviewsCount` ,`ActualPrice`,`OfferPrice` Columns to numerical columns

In [23]:
new_amazondf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1368 entries, 0 to 1367
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   PhoneModel     1368 non-null   object
 1   Rating         1368 non-null   object
 2   ReviewsCount   1368 non-null   object
 3   OfferPrice     1368 non-null   object
 4   ActualPrice    1368 non-null   object
 5   Saving         1368 non-null   object
 6   Delivery       1368 non-null   object
 7   StatusOfStock  1368 non-null   object
dtypes: object(8)
memory usage: 85.6+ KB


In [24]:
new_amazondf['ReviewsCount'] = new_amazondf['ReviewsCount'].astype('float').astype('int')
new_amazondf['OfferPrice'] = new_amazondf['OfferPrice'].astype('float').astype('int')
new_amazondf['ActualPrice'] = new_amazondf['ActualPrice'].astype('float').astype('int')



In [25]:
#This is the new data types after converting some columns from strings to integers
new_amazondf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1368 entries, 0 to 1367
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   PhoneModel     1368 non-null   object
 1   Rating         1368 non-null   object
 2   ReviewsCount   1368 non-null   int64 
 3   OfferPrice     1368 non-null   int64 
 4   ActualPrice    1368 non-null   int64 
 5   Saving         1368 non-null   object
 6   Delivery       1368 non-null   object
 7   StatusOfStock  1368 non-null   object
dtypes: int64(3), object(5)
memory usage: 85.6+ KB


<h3> Creating a new column for the % Saved</h3>

In [26]:
#Creating a new column for % saved 

new_amazondf['savings']= (new_amazondf['ActualPrice']- new_amazondf['OfferPrice'])/ new_amazondf['ActualPrice']
new_amazondf['savingsby%'] = (new_amazondf['savings'] *100)
new_amazondf.head().round(0)

PhoneModel              Rating  \
0  Redmi 9 (Sky Blue, 4GB RAM, 64GB Storage) | 2....  4.2 out of 5 stars   
1  Redmi 9A (Nature Green, 2GB RAM, 32GB Storage)...  4.2 out of 5 stars   
2  Redmi 9A (Sea Blue 2GB RAM 32GB Storage) | 2GH...  4.2 out of 5 stars   
3  Redmi Note 10 (Aqua Green, 4GB RAM, 64GB Stora...  4.2 out of 5 stars   
4  Redmi 9 (Carbon Black, 4GB RAM, 64GB Storage) ...  4.2 out of 5 stars   

   ReviewsCount  OfferPrice  ActualPrice             Saving  \
0         87809        9499        10999  Save ₹1,500 (14%)   
1         74587        6999         8499  Save ₹1,500 (18%)   
2         74587        6999         8499  Save ₹1,500 (18%)   
3         95519       13999        15999  Save ₹2,000 (13%)   
4         87809        9499        10999  Save ₹1,500 (14%)   

                  Delivery          StatusOfStock  savings  savingsby%  
0  FREE Delivery by Amazon  Only 1 left in stock.      0.0        14.0  
1  FREE Delivery by Amazon  Only 1 left in stock.      0.0        18.0  
2  FREE Delivery by Amazon  Only 1 left in stock.      0.0        18.0  
3  FREE Delivery by Amazon  Only 1 left in stock.      0.0        13.0  
4  FREE Delivery by Amazon  Only 1 left in stock.      0.0        14.0

In [27]:
#Dropping the columns i dont need
new_df= new_amazondf.drop(['Saving','savings'], axis = 1)
new_df.head().round(0)

PhoneModel              Rating  \
0  Redmi 9 (Sky Blue, 4GB RAM, 64GB Storage) | 2....  4.2 out of 5 stars   
1  Redmi 9A (Nature Green, 2GB RAM, 32GB Storage)...  4.2 out of 5 stars   
2  Redmi 9A (Sea Blue 2GB RAM 32GB Storage) | 2GH...  4.2 out of 5 stars   
3  Redmi Note 10 (Aqua Green, 4GB RAM, 64GB Stora...  4.2 out of 5 stars   
4  Redmi 9 (Carbon Black, 4GB RAM, 64GB Storage) ...  4.2 out of 5 stars   

   ReviewsCount  OfferPrice  ActualPrice                 Delivery  \
0         87809        9499        10999  FREE Delivery by Amazon   
1         74587        6999         8499  FREE Delivery by Amazon   
2         74587        6999         8499  FREE Delivery by Amazon   
3         95519       13999        15999  FREE Delivery by Amazon   
4         87809        9499        10999  FREE Delivery by Amazon   

           StatusOfStock  savingsby%  
0  Only 1 left in stock.        14.0  
1  Only 1 left in stock.        18.0  
2  Only 1 left in stock.        18.0  
3  Only 1 left in stock.        13.0  
4  Only 1 left in stock.        14.0

<h1>Statistical Analysis of Our Datasets</h1>

In [28]:
#This shows the statistical analysis of our numerical columns
new_df.describe()

ReviewsCount     OfferPrice    ActualPrice   savingsby%
count    1368.000000    1368.000000    1368.000000  1154.000000
mean     7095.711257   20663.812135   21161.931287         -inf
std     20938.652488   32780.345662   34788.356742          NaN
min         1.000000       0.000000       0.000000         -inf
25%         7.000000    1023.750000     999.000000     7.110866
50%       306.000000    9699.500000    5098.000000    14.372797
75%      2258.500000   21999.000000   23990.000000    25.685072
max    243715.000000  180890.000000  184800.000000    80.072793

In [29]:
#Getting the Average actual price value, offer value and review count of each phone model
print(f"Average Actual price for each phone is    : {new_df['ActualPrice'].mean()}")
print(f"Average Offer Value of each Phone : {new_df['OfferPrice'].mean()}")
print(f"Average reviews on each phone model : {new_df['ReviewsCount'].mean()}")

Average Actual price for each phone is    : 21161.93128654971
Average Offer Value of each Phone : 20663.812134502925
Average reviews on each phone model : 7095.711257309941


An Average Smartphone/Accesories sold in Amazon webisite was sold for ₹20,663 (₦113,235.87)while an average price value  of the phone was valued at ₹21,161 (₦115,964.98)

<h3>Which Phone Model/ Accessories has the highest number of Ratings?</h3

In [30]:
Min_Max_Rating= new_df.groupby('Rating')['PhoneModel'].agg(['min','max'])
Min_Max_Rating



min  \
Rating                                                                  
4.0 out of 5 stars  (Renewed) Honor View20 (Blue, 6GB RAM, 128GB S...   
4.1 out of 5 stars  (Renewed) OPPO Reno 2 (Luminous Black, 8GB RAM...   
4.2 out of 5 stars  (Renewed) OnePlus 9 5G (Winter Mist, 12GB RAM,...   
4.3 out of 5 stars                  (Renewed) Lava GEM (Gold + Black)   
4.4 out of 5 stars  (Renewed) Motorola G9 (Sapphire Blue, 64 GB) (...   
4.5 out of 5 stars    (Renewed) Nokia 3.4 Dusk, 4GB RAM, 64GB Storage   
4.6 out of 5 stars     (Renewed) Nokia C3 Sand, 3GB RAM, 32GB Storage   
4.7 out of 5 stars                  (Renewed) Nokia 2.1 (Blue/Copper)   
4.8 out of 5 stars                                   Lava A9 Dual SIM   
4.9 out of 5 stars  New Apple iPhone 12 (128GB) - Blue with Apple ...   
5.0 out of 5 stars                   (Renewed) Lava 34ULTRA Teal Blue   

                                                                  max  
Rating                                                                 
4.0 out of 5 stars  realme narzo 30 (Racing Sliver, 6GB RAM, 128GB...  
4.1 out of 5 stars           realme X7 Max 5G (Milky Way, 12GB+256GB)  
4.2 out of 5 stars  realme X7 Max 5G (Mercury Silver, 8GB RAM, 128...  
4.3 out of 5 stars     realme XT (Pearl White, 6GB RAM, 64GB Storage)  
4.4 out of 5 stars           itel Muzik110 (4.5cm, Loud Sound, Black)  
4.5 out of 5 stars     itel it2192T(4.5cm, Thermo Edition, Deep Blue)  
4.6 out of 5 stars      itel A42 Plus 2GB RAM With Fingerprint Unlock  
4.7 out of 5 stars                itel it5626 (7.1cm, 2500mAh, Black)  
4.8 out of 5 stars  Nillkin Samsung Galaxy S21 Case, CamShield Pro...  
4.9 out of 5 stars  Nillkin Super Frosted Shield Hard Back Cover C...  
5.0 out of 5 stars  nillkin tempered glass for samsung galaxy note...

Nikin Tempered Glass for Samsung Galaxy Note is the top rated accessory( 5.0  star ratings ), followed closely with New Apple iphone 12(128GB) with 4.9 star 

<h3>What is the value count of ratings?</h3>

In [31]:
#value count of Rating
new_df['Rating'].value_counts()

4.2 out of 5 stars    279
4.1 out of 5 stars    234
4.0 out of 5 stars    228
5.0 out of 5 stars    185
4.6 out of 5 stars    140
4.3 out of 5 stars    128
4.4 out of 5 stars     69
4.5 out of 5 stars     57
4.7 out of 5 stars     31
4.9 out of 5 stars     13
4.8 out of 5 stars      4
Name: Rating, dtype: int64

The rating range lies from 4.1 to 5.0. However 4.2 star rating has the highest number of ratings(279) while 4.8 had the least rating (4).This means Amazon has high reputation score and customers are statisfied with the phone accessories and product they purchased from the e commerce site

<h3>Which Phone Model/Accessory is the highest review Count</h3>

In [32]:
Min_Max_ReviewsCount= new_df.groupby('ReviewsCount')['PhoneModel'].agg(['min','max'])
Min_Max_ReviewsCount

min  \
ReviewsCount                                                      
1                              (Renewed) Lava 34ULTRA Teal Blue   
2               (Renewed) Nokia 3.4 Dusk, 4GB RAM, 64GB Storage   
3             (Renewed) Mi 10T 5G (Cosmic Black, 8GB RAM, 12...   
4             (Renewed) Nokia 5.4 (Polar Night, 4GB RAM, 64G...   
5                             (Renewed) Lava GEM (Gold + Black)   
...                                                         ...   
103817        Redmi Note 9 (Pebble Grey, 4GB RAM 64GB Storag...   
104653        Redmi Note 9 Pro (Glacier White, 4GB RAM, 64GB...   
125853        Redmi Note 8 Pro (Shadow Black, 8GB RAM, 128GB...   
170469        Samsung Galaxy M21 2021 Edition (Arctic Blue, ...   
243715        Samsung Galaxy M31 (Ocean Blue, 8GB RAM, 128GB...   

                                                            max  
ReviewsCount                                                     
1             nillkin tempered glass for samsung galaxy note...  
2             nillkin qin series luxury leather wallet Flip ...  
3                     hasija Selling micromax x741 (Mix Colour)  
4                           itel it5626 (7.1cm, 2500mAh, Black)  
5             Ziox Z351 with Vibration 2.4"" Display Wireles...  
...                                                         ...  
103817         Redmi Note 9 Shadow Black, 4GB RAM, 64GB Storage  
104653        Redmi Note 9 Pro (Glacier White, 4GB RAM, 64GB...  
125853        Redmi Note 8 Pro (Shadow Black, 8GB RAM, 128GB...  
170469        Samsung Galaxy M21 2021 Edition (Charcoal blac...  
243715        Samsung Galaxy M31 (Space Black, 8GB RAM, 128G...  

[363 rows x 2 columns]

Nillkin tempered glass screen protector has the most review count

<h3>Which product has the highest rating and highest %savings</h3>

In [33]:
# sorting the dataset to get rows with scores with 5 point star rating,  with over 26% discount & at least one review
new_df.loc[(new_df['Rating'] == '5.0 out of 5 stars') & (new_df['savingsby%'] >26) & (new_df['ReviewsCount']>1)]


PhoneModel              Rating  \
593   Nillkin Case for Samsung Galaxy S10 S 10 Lite ...  5.0 out of 5 stars   
598   Nillkin Slim Frosted Anti-Falling Anti Fingerp...  5.0 out of 5 stars   
600   Nillkin Case for Samsung Galaxy S10 S 10 Lite ...  5.0 out of 5 stars   
624   Nillkin Slim Frosted Anti-Falling Anti Fingerp...  5.0 out of 5 stars   
654   Nillkin Case for OnePlus 9 Pro One Plus 9 Pro ...  5.0 out of 5 stars   
746   (Renewed) Mi 10T 5G (Cosmic Black, 8GB RAM, 12...  5.0 out of 5 stars   
825   IAIR Basic Feature Dual Sim Mobile Phone with ...  5.0 out of 5 stars   
842   Nillkin Frosted Shield Ultra Thin Hard Plastic...  5.0 out of 5 stars   
865   Stealodeal Virus Protection No Touch Stylus Ca...  5.0 out of 5 stars   
921   Nillkin for Samsung Galaxy S21 Plus Case, CamS...  5.0 out of 5 stars   
939   Nillkin for Samsung Galaxy S21 Plus Case, CamS...  5.0 out of 5 stars   
943   Kechaoda A32 Bluetooth Dialer 1.44 inch Dual S...  5.0 out of 5 stars   
1023  Nillkin Super Frosted Shield Ultra Thin Hard P...  5.0 out of 5 stars   
1119  Nillkin Frosted Shield Ultra Thin Hard Plastic...  5.0 out of 5 stars   
1190  Nillkin Case for Samsung Galaxy S21 Ultra S 21...  5.0 out of 5 stars   
1202  Nillkin Textured Nylon Fiber Luxury Look and G...  5.0 out of 5 stars   
1216  Kalra Pubg Trigger Controller, Mobile Gamepad ...  5.0 out of 5 stars   
1223  Nillkin Super Frosted Shield Pro Magnetic Hard...  5.0 out of 5 stars   
1227  Nillkin Super Frosted Shield Pro Magnetic Hard...  5.0 out of 5 stars   
1237  Samsung Galaxy S10 Plus Ceramic White, 12GB RA...  5.0 out of 5 stars   
1284  MobiSpare® Orignal LCD Display with Touch Scre...  5.0 out of 5 stars   
1288  Amazing H+ PRO Anti-Explosion Tempered Glass S...  5.0 out of 5 stars   
1324  Nillkin Case for Huawei Mate 30 Pro/Mate 30 Pr...  5.0 out of 5 stars   
1341  Nillkin Case for Samsung Galaxy J6 Plus/J 6 Pr...  5.0 out of 5 stars   
1350  VRS Design Case for Apple iPhone 8 Plus (5.5" ...  5.0 out of 5 stars   

      ReviewsCount  OfferPrice  ActualPrice  \
593              2         743         1499   
598              6         525         1999   
600              2         743         1499   
624              6         525         1999   
654              3         699         1999   
746              3       25249        42999   
825              2        1199         1799   
842              3         850         1990   
865              2         169          299   
921              6        1049         2999   
939              6        1049         2999   
943              3         996         1499   
1023             2         575         1299   
1119             2         750         1990   
1190             2        1999         2999   
1202             2        1499         2999   
1216             2         449          999   
1223             2        1799         3999   
1227             2        1799         3999   
1237             5       62990       124990   
1284             3        1999         3999   
1288             2         879         1699   
1324             2        1199         1999   
1341             2         899         1299   
1350             3        1299         3499   

                                               Delivery  \
593                             FREE Delivery by Amazon   
598                             FREE Delivery by Amazon   
600                             FREE Delivery by Amazon   
624                             FREE Delivery by Amazon   
654                             FREE Delivery by Amazon   
746                             FREE Delivery by Amazon   
825                             FREE Delivery by Amazon   
842                             FREE Delivery by Amazon   
865   FREE Delivery on your first order in this cate...   
921                             FREE Delivery by Amazon   
939                             FREE Delivery by Amazon   
943                             FREE Delivery by Amazon   


Nillkin Slim Frosted Anti-Falling Anti Fingerprint is the best selling product. This can be because it has the highest discount rate too